### PR TITLE
[APPS-2019] Add Undeletable and Unmovable Aspects to Data Dictionary and its Level 1 sub-folders

### PR DESCRIPTION
**Pull request by suneet-gupta**
https://hyland.atlassian.net/browse/APPS-2019

### CODE REVIEW

File: repository/src/test/java/org/alfresco/repo/bootstrap/DataDictionaryFolderTest.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        AuthenticationUtil.setFullyAuthenticatedUser(AuthenticationUtil.getAdminUserName());
+        // get the company_home
+        NodeRef companyHomeRef = wellKnownNodes.getCompanyHome();
+        // get the Data Dictionary
+        NodeRef dataDictionaryRef = nodeService.getChildByName(companyHomeRef, ContentModel.ASSOC_CONTAINS,
+                "Data Dictionary");
+
+        List<ChildAssociationRef> chilAssocsList = nodeService.getChildAssocs(dataDictionaryRef);
+
+        AuthenticationUtil.setFullyAuthenticatedUser(AuthenticationUtil.getAdminUserName());
```
Comment (purusothaman-mm) [inline repository/src/test/java/org/alfresco/repo/bootstrap/DataDictionaryFolderTest.java:75]: You have already added the authentication in line 66. But duplication occurs in line 73.

**Review by code4uuuu — APPROVED** [review]
LGTM!

File: repository/src/test/java/org/alfresco/repo/bootstrap/DataDictionaryFolderTest.java (hunk 1 above)
Comment (suneet-gupta) [reply repository/src/test/java/org/alfresco/repo/bootstrap/DataDictionaryFolderTest.java]: yes..removed the duplicated line

**Review by purusothaman-mm — APPROVED** [review]
LGTM

**Review by mohit-singh4 — APPROVED** [review]
LGTM!

File: repository/src/test/java/org/alfresco/repo/bootstrap/DataDictionaryFolderTest.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    {
+        AuthenticationUtil.setFullyAuthenticatedUser(AuthenticationUtil.getAdminUserName());
+        // get the company_home
+        NodeRef companyHomeRef = wellKnownNodes.getCompanyHome();
+        // get the Data Dictionary
+        NodeRef dataDictionaryRef = nodeService.getChildByName(companyHomeRef, ContentModel.ASSOC_CONTAINS, "Data Dictionary");
+
+        List<ChildAssociationRef> chilAssocsList = nodeService.getChildAssocs(dataDictionaryRef);
+
+        for (ChildAssociationRef childAssociationRef : chilAssocsList)
```
Comment (kavitshah-gl) [inline repository/src/test/java/org/alfresco/repo/bootstrap/DataDictionaryFolderTest.java:74]: use Java 8 features.

File: repository/src/test/java/org/alfresco/repo/bootstrap/DataDictionaryFolderTest.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+    @Test
+    public void testDataDictionaryFolderIsUndeletable()
+    {
+        AuthenticationUtil.setFullyAuthenticatedUser(AuthenticationUtil.getAdminUserName());
+        // get the company_home
+        NodeRef companyHomeRef = wellKnownNodes.getCompanyHome();
+        // get the Data Dictionary
+        NodeRef dataDictionaryRef = nodeService.getChildByName(companyHomeRef, ContentModel.ASSOC_CONTAINS, "Data Dictionary");
```
Comment (kavitshah-gl) [inline repository/src/test/java/org/alfresco/repo/bootstrap/DataDictionaryFolderTest.java:70]: String should be final and static

**Review by kavitshah-gl — CHANGES_REQUESTED** [review]

**Comment by suneet-gupta** [comment]
> Can we add the test cases to delete and move the folder? If there are already test cases which are testing these aspects then we shall reference them otherwise we shall have rest assured test cases for them.

Added the test cases as per the comment

File: repository/src/test/java/org/alfresco/repo/bootstrap/DataDictionaryFolderTest.java (hunk 3 above)
Comment (suneet-gupta) [reply repository/src/test/java/org/alfresco/repo/bootstrap/DataDictionaryFolderTest.java:72]: Changes done as per the comment

File: repository/src/test/java/org/alfresco/repo/bootstrap/DataDictionaryFolderTest.java (hunk 2 above)
Comment (suneet-gupta) [reply repository/src/test/java/org/alfresco/repo/bootstrap/DataDictionaryFolderTest.java]: Changes done as per the comment

**Review by kavitshah-gl — APPROVED** [review]
Approved

**Pull request merged**